### PR TITLE
[SMALLFIX] Use InetAddress.getLocalHost() instead of hardcoded localhost

### PR DESCRIPTION
**Pull request by yupeng9**
This is better, because 'locahost' might not always be resolved to the correct IP address, as in virtual network.

### CODE REVIEW

File: common/src/test/java/tachyon/security/authentication/AuthenticationUtilsTest.java [hunk 1]
```diff
@@ -58,9 +59,9 @@
   public void before() throws Exception {
     mTachyonConf = new TachyonConf();
     // Use port 0 to assign each test case an available port (possibly different)
-    mServerTSocket = new TServerSocket(new InetSocketAddress("localhost", 0));
+    mServerTSocket = new TServerSocket(new InetSocketAddress(InetAddress.getLocalHost(), 0));
     int port = NetworkAddressUtils.getThriftPort(mServerTSocket);
-    mServerAddress = new InetSocketAddress("localhost", port);
+    mServerAddress = new InetSocketAddress(InetAddress.getLocalHost(), port);
```
Comment (calvinjia) [inline common/src/test/java/tachyon/security/authentication/AuthenticationUtilsTest.java]: Should the utility in `NetworkAddressUtils` be used?

Comment (yupeng9) [inline common/src/test/java/tachyon/security/authentication/AuthenticationUtilsTest.java]: Good to know. Done.

**Comment by apc999** [comment]
+1 LGTM. Thanks for fixing the tests in virtual net!

**Comment by calvinjia** [comment]
LGTM

**Pull request merged**